### PR TITLE
feat/dependency-version-condition-12 

### PR DESCRIPTION
**Pull request by bnenu**
Added  condition validation when a dependency in package json matches a certain value

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,6 +9,13 @@
 // TODO::after_date("2021-04-02"): remove april fools code
 ```
 
+```typescript
+// Will result in your editor showing an error: "Your package has arrived! now
+on 4.5.1"
```
Comment (ngnijland) [inline README.md:14]: You should put this at the end of line 13. Otherwise it won't show as a comment

File: src/types.ts [hunk 2]
```diff
@@ -17,8 +18,27 @@ export type ConfigOptions = {
...
+};
+
+export type WhenConfig = {
+  pjson?: Record<string, any> | null;
+  options?: ConfigOptions["when"];
+};
+
+export type ValidateMetaData = {
+  options: ConfigOptions;
+  packageJson: Record<string, any> | null;
```
Comment (ngnijland) [inline src/types.ts:33]: Can we type this as `unknown` instead of `any`?

File: src/utils.ts [hunk 3]
```diff
@@ -1,32 +1,57 @@
...
+  const [year] = date.split("-");
+  return year.length === 4;
+};
 
 export const isValidWarnOption = (option: string): string | boolean => {
   return option.match(/^\d+[w|d|h]$/) ? option : false;
 };
 
+export const isValidWarnWhenOption = (option: string): string | boolean => {
+  return option.match(/^\d+[M|m|p]$/) ? option : false;
```
Comment (ngnijland) [inline src/utils.ts:29]: I think it will be a bit more clear if this function just returns a boolean. What do you think?

File: src/utils.test.ts [hunk 4]
```diff
@@ -1,85 +1,155 @@
...
+  const option1 = true
+  const option2 = '2m'
+
+  expect(getWarningWhen(option1)).toStrictEqual([1, 2])
+  expect(getWarningWhen(option2)).toStrictEqual([2, 1])
+})
+
+test('getSplitPackageVersion', () => {
+  const p = '1.2.3'
+  expect(getSplitPackageVersion(p)).toStrictEqual([1, 2, 3])
```
Comment (ngnijland) [inline src/utils.test.ts:145]: Did you take into account that people cab also use `2.x` notations for example in their `package.json` file?

Comment (bnenu) [reply src/utils.test.ts:145]: No, not really! It was quite a ride to get this checking right with the full version description! I think we can have that as an extension!

Comment (ngnijland) [reply src/utils.test.ts:145]: Fair point! I'll create an issue from this comment.

Maybe we can checkout the source code of this elixir credo plugin to check how it's done there!
https://hex.pm/packages/credo_todo_or_die

**Review by ngnijland — APPROVED** [review]

**Pull request merged**